src/plugins/discover/public/application/angular/discover.js

```diff
@@ -940,6 +942,8 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
     const { indexPattern, searchSource } = $scope;
     searchSource
       .setField('index', $scope.indexPattern)
+      .setField('fieldsApi', ['*'])
+      .setField('source', false)
```

Comment:
@majagrubic all fields are requested instead of source

Reply:
so it's the opposite of https://github.com/elastic/kibana/pull/75407 where _source is requested and selected fields are added to the requests, here no additional request is needed when columns are added to the table